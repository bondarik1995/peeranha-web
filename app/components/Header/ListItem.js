import styled from 'styled-components';

const ListItem = styled.li`
  padding: 8px 8px 0 12px;
  font-size: 18px;
  @media only screen and (min-width: 992px) {
    &:first-child {
      padding-left: 32px;
    }
  }
`;

export default ListItem;
