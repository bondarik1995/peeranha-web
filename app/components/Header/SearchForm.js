import styled from 'styled-components';

const SearchForm = styled.form`
  flex-wrap: nowrap;
  @media only screen and (max-width: 576px) {
    * {
      margin: 0 2px;
    }
  }
`;

export default SearchForm;
