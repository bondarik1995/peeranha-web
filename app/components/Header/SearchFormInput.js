import styled from 'styled-components';

const SearchFormInput = styled.input`
  flex: 1;
`;

export default SearchFormInput;
