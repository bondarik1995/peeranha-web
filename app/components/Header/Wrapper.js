import styled from 'styled-components';

const Wrapper = styled.header`
  padding: 10px 0;
`;

export default Wrapper;
