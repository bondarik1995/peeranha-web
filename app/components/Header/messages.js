/*
 * Header Messages
 *
 * This contains all the text for the Header component.
 */

import { defineMessages } from 'react-intl';

export default defineMessages({
  home: {
    id: 'app.components.Header.home',
    defaultMessage: 'Home',
    mark: 'tab',
    link: '/',
    tabCode: 1,
  },
  profile: {
    id: 'app.components.Header.profile',
    defaultMessage: 'Profile',
    mark: 'tab',
    link: '/profile',
    tabCode: 2,
  },
  about: {
    id: 'app.components.Header.about',
    defaultMessage: 'About',
    mark: 'tab',
    link: '/about',
    tabCode: 3,
  },
  help: {
    id: 'app.components.Header.help',
    defaultMessage: 'Help',
    mark: 'tab',
    link: '/help',
    tabCode: 4,
  },
  search: {
    id: 'app.components.Header.search',
    defaultMessage: 'Search',
  },
  signIn: {
    id: 'app.components.Header.signIn',
    defaultMessage: 'Sign In',
  },
});
