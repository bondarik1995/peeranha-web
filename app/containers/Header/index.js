/**
 *
 * Header
 *
 */

import React from 'react';
import { connect } from 'react-redux';
import { createStructuredSelector } from 'reselect';
import { compose } from 'redux';
import { Link } from 'react-router-dom';

import injectSaga from 'utils/injectSaga';
import injectReducer from 'utils/injectReducer';
import makeSelectHeader from './selectors';
import reducer from './reducer';
import saga from './saga';

import Wrapper from './styledComponents/wrapper';
import Logo from './styledComponents/logo';
import ListItem from './styledComponents/listItem';
import SearchForm from './styledComponents/searchForm';
import SearchFormInput from './styledComponents/searchFormInput';

/* eslint-disable react/prefer-stateless-function */
export class Header extends React.Component {
  /* 
  * signFunction method handles sign operations 
  */

  signFunction(e) {
    e.preventDefault();
    console.log('signFunction');
  }

  render() {
    return (
      <Wrapper className="col-xl-8 col-lg-10">
        <nav className="navbar navbar-expand-lg navbar-light">
          <Link to="/" href="/">
            <Logo />
          </Link>

          <button
            className="navbar-toggler"
            type="button"
            data-toggle="collapse"
            data-target="#navbarTogglerDemo02"
            aria-controls="navbarTogglerDemo02"
            aria-expanded="false"
            aria-label="Toggle navigation"
          >
            <span className="navbar-toggler-icon" />
          </button>

          <div className="collapse navbar-collapse" id="navbarTogglerDemo02">
            <ul className="navbar-nav mr-auto mt-2 mt-lg-0">
              <ListItem className="nav-item">
                <Link to="/" className="nav-link" href="/">
                  Home
                </Link>
              </ListItem>

              <ListItem className="nav-item">
                <Link to="/profile" className="nav-link" href="/profile">
                  Profile
                </Link>
              </ListItem>

              <ListItem className="nav-item">
                <Link to="/about" className="nav-link" href="/about">
                  About
                </Link>
              </ListItem>

              <ListItem className="nav-item">
                <Link to="/help" className="nav-link" href="/help">
                  Help
                </Link>
              </ListItem>
            </ul>

            <SearchForm className="form-inline my-2 my-lg-2">
              <SearchFormInput
                className="form-control mr-sm-2"
                type="search"
                placeholder="Search"
              />
              <button
                className="btn btn-outline-success mr-sm-2 my-2 my-sm-0"
                type="submit"
              >
                Search
              </button>
              <button
                className="btn btn-success my-2 my-sm-0"
                onClick={e => this.signFunction(e)}
              >
                Sign In
              </button>
            </SearchForm>
          </div>
        </nav>
      </Wrapper>
    );
  }
}

Header.propTypes = {};

const mapStateToProps = createStructuredSelector({
  header: makeSelectHeader(),
});

function mapDispatchToProps(dispatch) {
  return {
    dispatch,
  };
}

const withConnect = connect(
  mapStateToProps,
  mapDispatchToProps,
);

const withReducer = injectReducer({ key: 'header', reducer });
const withSaga = injectSaga({ key: 'header', saga });

export default compose(
  withReducer,
  withSaga,
  withConnect,
)(Header);
