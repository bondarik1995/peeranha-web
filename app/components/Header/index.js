/**
 *
 * Header
 *
 */

import React from 'react';
import { Link } from 'react-router-dom';

import Wrapper from './Wrapper';
import Logo from './Logo';
import ListItem from './ListItem';
import SearchForm from './SearchForm';
import SearchFormInput from './SearchFormInput';

import messages from './messages';

/* eslint-disable react/prefer-stateless-function */
const Header = () => {
  const MSG_KEYS = Object.keys(messages);
  return (
    <Wrapper className="col-xl-8 col-lg-10">
      <nav className="navbar navbar-expand-lg navbar-light">
        <Link to="/" href="/">
          <Logo />
        </Link>

        <button
          className="navbar-toggler"
          type="button"
          data-toggle="collapse"
          data-target="#navbarTogglerDemo02"
          aria-controls="navbarTogglerDemo02"
          aria-expanded="false"
          aria-label="Toggle navigation"
        >
          <span className="navbar-toggler-icon" />
        </button>

        <div className="collapse navbar-collapse" id="navbarTogglerDemo02">
          <ul className="navbar-nav mr-auto mt-2 mt-lg-0">
            {MSG_KEYS.map(
              item =>
                messages[item].mark === 'tab' && (
                  <ListItem className="nav-item" key={messages[item].tabCode}>
                    <Link
                      to={messages[item].link}
                      className="nav-link"
                      href={messages[item]}
                    >
                      {messages[item].defaultMessage}
                    </Link>
                  </ListItem>
                ),
            )}
          </ul>

          <SearchForm className="form-inline my-2 my-lg-2">
            <SearchFormInput
              className="form-control mr-sm-2"
              type="search"
              placeholder={messages.search.defaultMessage}
            />
            <button
              className="btn btn-outline-success mr-sm-2 my-2 my-sm-0"
              type="submit"
            >
              {messages.search.defaultMessage}
            </button>
            <Link to="/sign" href="/sign">
              <button className="btn btn-success my-2 my-sm-0">
                {messages.signIn.defaultMessage}
              </button>
            </Link>
          </SearchForm>
        </div>
      </nav>
    </Wrapper>
  );
};

export default Header;
