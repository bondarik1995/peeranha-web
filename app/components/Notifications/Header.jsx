import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import { FormattedMessage } from 'react-intl';

import messages from 'common-messages';

import {
  BG_SECONDARY_LIGHT,
  BORDER_SECONDARY,
  TEXT_SECONDARY_LIGHT,
} from 'style-constants';

import notificationsIcon from 'images/Notifications_Bright.svg?inline';

import H3 from '../H3';
import Span from '../Span';

const Container = styled.div`
  display: flex;
  align-items: center;
  > h3 {
    margin-right: 20px;
  }
`;

const Icon = styled.div`
  width: 43px;
  height: 43px;
  min-width: 43px;
  display: flex;
  align-items: center;
  justify-content: center;
  background: ${BG_SECONDARY_LIGHT};
  border: 1px solid ${BORDER_SECONDARY};
  border-radius: 50%;
  margin-right: 20px;
`;

const Header = ({ notificationsNumber }) => (
  <Container>
    <Icon>
      <img src={notificationsIcon} width="25" alt="notifications_icon" />
    </Icon>
    <H3>
      <FormattedMessage {...messages.messageCenter} />
    </H3>
    <Span
      fontSize="38"
      lineHeight="48"
      color={TEXT_SECONDARY_LIGHT}
      bold
      whiteSpace="nowrap"
    >
      {notificationsNumber}
    </Span>
  </Container>
);

Header.propTypes = {
  notificationsNumber: PropTypes.number.isRequired,
};

export default Header;
