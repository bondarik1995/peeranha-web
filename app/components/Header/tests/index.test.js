import React from 'react';
import { BrowserRouter } from 'react-router-dom';
import { IntlProvider } from 'react-intl';
import renderer from 'react-test-renderer';
import 'jest-styled-components';

import Header from '../index';

describe('<Header />', () => {
  it('should match the snapshot', () => {
    const renderedComponent = renderer
      .create(
        <BrowserRouter>
          <IntlProvider locale="en">
            <Header />
          </IntlProvider>
        </BrowserRouter>,
      )
      .toJSON();
    expect(renderedComponent).toMatchSnapshot();
  });
});
