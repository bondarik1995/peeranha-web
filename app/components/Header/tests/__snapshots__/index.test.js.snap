// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> should match the snapshot 1`] = `
.c0 {
  padding: 10px 0;
}

.c1 {
  height: 50px;
  width: 160px;
  background: url(0:I; 1:M; 2:A; 3:G; 4:E; 5:_; 6:M; 7:O; 8:C; 9:K; default:IMAGE_MOCK;) no-repeat;
  background-size: cover;
  cursor: pointer;
  border-radius: 3px;
  padding: 25px;
}

.c2 {
  padding: 8px 8px 0 12px;
  font-size: 18px;
}

.c3 {
  -webkit-flex-wrap: nowrap;
  -ms-flex-wrap: nowrap;
  flex-wrap: nowrap;
}

.c4 {
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
}

@media only screen and (min-width:992px) {
  .c2:first-child {
    padding-left: 32px;
  }
}

@media only screen and (max-width:576px) {
  .c3 * {
    margin: 0 2px;
  }
}

<header
  className="c0"
>
  <nav
    className="container navbar navbar-expand-lg navbar-light"
  >
    <a
      href="/"
      onClick={[Function]}
    >
      <img
        className="c1"
      />
    </a>
    <button
      aria-controls="navbarTogglerDemo02"
      aria-expanded="false"
      aria-label="Toggle navigation"
      className="navbar-toggler"
      data-target="#navbarTogglerDemo02"
      data-toggle="collapse"
      type="button"
    >
      <span
        className="navbar-toggler-icon"
      />
    </button>
    <div
      className="collapse navbar-collapse"
      id="navbarTogglerDemo02"
    >
      <ul
        className="navbar-nav mr-auto mt-2 mt-lg-0"
      >
        <li
          className="nav-item c2"
        >
          <a
            className="nav-link"
            href="/"
            onClick={[Function]}
          >
            <span>
              Home
            </span>
          </a>
        </li>
        <li
          className="nav-item c2"
        >
          <a
            className="nav-link"
            href="/profile"
            onClick={[Function]}
          >
            <span>
              Profile
            </span>
          </a>
        </li>
        <li
          className="nav-item c2"
        >
          <a
            className="nav-link"
            href="/about"
            onClick={[Function]}
          >
            <span>
              About
            </span>
          </a>
        </li>
        <li
          className="nav-item c2"
        >
          <a
            className="nav-link"
            href="/help"
            onClick={[Function]}
          >
            <span>
              Help
            </span>
          </a>
        </li>
      </ul>
      <form
        className="form-inline my-2 my-lg-2 c3"
      >
        <input
          className="form-control mr-sm-2 c4"
          placeholder="Search"
          type="search"
        />
        <button
          className="btn btn-outline-success mr-sm-2 my-2 my-sm-0"
          type="submit"
        >
          <span>
            Search
          </span>
        </button>
        <a
          href="/sign"
          onClick={[Function]}
        >
          <button
            className="btn btn-success my-2 my-sm-0"
          >
            <span>
              Sign In
            </span>
          </button>
        </a>
      </form>
    </div>
  </nav>
</header>
`;
